### ISSUE TITLE
Major glitches when you try to change fullscreen status when the carousel is moving

### ISSUE THREAD

**Issue by nkalupahana** [metafizzy/flickity-fullscreen#32]
If you go the [demo](https://flickity.metafizzy.co/options.html#fullscreen), click the next button on the carousel, and then immediately click the fullscreen button, instead of seeing the fullscreen carousel, you'll see a black image. Similarly, if you drag the carousel while it's in fullscreen and then click the button to exit, the carousel image will disappear.

This bug makes this functionality unusable, and any insight into why it might be happening would be greatly appreciated.

**Comment by AndreasDeRijcke** [comment]
I confirm both the issue and the patch having the desired result.

**Comment by txrris** [comment]
This is NOT fixed

**Comment by nkalupahana** [comment]
@txrris Yes, it is not -- that's what PR #33 is for, which hopefully will eventually get merged :/

**Comment by Alexandre-dcm** [comment]
Thx a lot for the fix @nkalupahana

**Comment by mattneal-stafflink** [comment]
Any updates on this? The problem still exists.

**Comment by indigoxela** [comment]
Unfortunately the original pull request has conflicts. And I'm not sure if it's a good idea to declare the whole function async, but a Promise is definitely the way to go.

This solution works for me: https://github.com/metafizzy/flickity-fullscreen/pull/44
It wraps functionality in a then() and uses the API ('settle' event) to figure out when the animation stopped, if there was one.

Seems like @desandro isn't following this issue queue too closely. But eventually this pull request triggers some attention. 😉
